pandas/core/ops/docstrings.py

```diff
@@ -1,7 +1,19 @@
 """
 Templating for ops docstrings
 """
-from typing import Dict, Optional
+from typing import TYPE_CHECKING, Dict, Optional
+
+if TYPE_CHECKING:
+    from mypy_extensions import TypedDict
```

Comment:
Is this a hard dependency of a third party package now?

Reply:
is this the one that lets us use "Literal"?

Reply:
> Is this a hard dependency of a third party package now?

only if type checking without mypy.

TypedDict is part of the standard library from py3.8

mypy_extensions is part of mypy. if we want to switch to pyre could import from typing_extensions instead.

> is this the one that lets us use "Literal"?

that's in typing_extensions, which is a required dependency of mypy in 0.730

```
     install_requires=['typed_ast >= 1.4.0, < 1.5.0',
                        'typing_extensions>=3.7.4',
                        'mypy_extensions >= 0.4.3, < 0.5.0',
                        ],
```

Reply:
I'm +/-0 on this. So definitely some nice stuff, but I guess since `mypy_extensions` is all experimental maybe some downsides in reaching out to that. 

I don't have personal experience with it so will defer

Reply:
I'll close for now. can always reopen at a later date.